amlb/runners/docker.py

```diff
@@ -51,9 +51,17 @@ def _start_container(self, script_params=""):
             touch(d, as_dir=True)
         script_extra_params = ""
         inst_name = self.sid
+
+        if "max_mem_size_mb" in self.constraint_def:
+            memory = "--memory={}m".format(self.constraint_def['max_mem_size_mb'])
+            memory += " " + "--memory-swap={}g".format(2 * float(self.constraint_def['max_mem_size_mb']) / 1024)
```

Comment:
for the swap, I would default it to -1 and suggest to offer the possibility to override this thanks to a property in the `docker` namespace in `config.yaml§ as I don't see why your `2 * max_mem` estimation would generalize.

Reply:
Some memory should be available for other system processes, so memory in container must be set with higher value than limit for benchmark. In the case of full memory, this can cause the process to be killed by the OOM killer, I faced this problem several times due running the benchmarks. So I think we should set the default swap value greater than 0 or actual container limits should be set with a higher value than the limit for benchmark.

Reply:
If I undestand docker's doc above correctly, `0` doesn't mean `no-swap` but `unset`, and:
```
if --memory="300m" and --memory-swap is not set, the container can use 600m in total of memory and swap.
```

Reply:
> or actual container limits should be set with a higher value than the limit for benchmark

That's why I suggest that it should be possible to override those docker memory limits through the `config.yaml` property.

Reply:
@sebhrusen ok, got it 👍 

Also I want to add `max_mem_size_mb` setting to resources/constraints.yaml like this:
```
1h8c32g:
  folds: 10
  max_runtime_seconds: 3600
  cores: 8
  min_vol_size_mb: 16384
  max_mem_size_mb: 32768
```
There aren't that many different configurations, so for now, it looks ok, what do you think?

Reply:
I support that: was also recently surprised that we forgot to add it there.

When no provided, it should default to all available memory.
Thanks for your help on this.